src/hooks/useAuthContext.js

```diff
@@ -0,0 +1,12 @@
+import { useContext } from 'react';
+import { AuthContext } from '../contexts/AuthContext';
+
+export const useAuthContext = () => {
+  const context = useContext(AuthContext);
+
+  if (!context) {
+    throw Error('useAuthContext must be used inside an AuthContextProvider');
```

Comment:
Who will catch this error if thrown? How we will handle this?

Reply:
It's for development mode only!